domain/src/main/java/io/homeassistant/companion/android/domain/integration/IntegrationUseCase.kt

```diff
@@ -3,6 +3,7 @@ package io.homeassistant.companion.android.domain.integration
 interface IntegrationUseCase {
 
     suspend fun registerDevice(deviceRegistration: DeviceRegistration)
+    suspend fun updateRegistration(deviceRegistration: DeviceRegistration)
```

Comment:
Not all fields can be updated. From top of my head, I think we can't update encryption and maybe app id?

Reply:
[Correct](https://developers.home-assistant.io/docs/en/app_integration_sending_data.html#update-registration).

Only `app_data`, `app_version`, `device_name`, `manufacturer`, `model` and `os_version`.

Reply:
I'll update the usecase to only have those fields, and reuse the objects on the backend side.